spec/routing/media_routing_spec.rb

```diff
@@ -6,23 +6,111 @@
       'media#download', id: 'oo000oo0000', file_name: 'aa666aa1234', format: 'mp4')
   end
 
-  it 'download: filename with hyphen' do
-    expect(get: '/media/oo000oo0000/aa666aa1234-a.mp4').to route_to(
-      'media#download', id: 'oo000oo0000', file_name: 'aa666aa1234-a', format: 'mp4')
-  end
+  context '#download: filename with' do
+    it 'hyphen' do
+      expect(get: '/media/oo000oo0000/aa666aa1234-a.mp4').to route_to(
+        'media#download', id: 'oo000oo0000', file_name: 'aa666aa1234-a', format: 'mp4')
+    end
+
+    it 'chars not requiring URI escaping' do
+      # filename = "(no_escape_needed):;=&$*.-_+!,~'" # can't use dot here, see applicable routing constraint
+      filename = "(no_escape_needed):;=&$*-_+!,~'"
+      expect(get: "/media/oo000oo0000/#{filename}.mp4").to route_to(
+        'media#download', id: 'oo000oo0000', file_name: filename, format: 'mp4')
+    end
+
+    it 'some chars requiring URI escaping' do
+      filename = 'escape_needed {} @#^ %|"`'
+      expect(get: "/media/oo000oo0000/#{URI.escape(filename)}.mp4").to route_to(
+        'media#download', id: 'oo000oo0000', file_name: filename, format: 'mp4')
+    end
+
+    it 'square brackets must be url escaped' do
+      skip("problem writing test:  the route doesn't seem to match on brackets, even if they're escaped")
```

Comment:
i'm not sure, and also found this behavior surprising.  maybe the regex in the routes config isn't escaping the brackets correctly?  the regex for the show route accepts anything that's not a slash, whereas the regex for download and stream explicitly lists all of the characters it find acceptable.  maybe `\[` doesn't actually match a bracket.